src/main/kotlin/no/nav/tilleggsstonader/sak/ekstern/journalføring/AutomatiskJournalføringService.kt

```diff
@@ -0,0 +1,218 @@
+        val personIdent = automatiskJournalføringRequest.personIdent
+        val stønadstype = automatiskJournalføringRequest.stønadstype
+
+        return if (kanOppretteBehandling(personIdent, stønadstype)) {
+            automatiskJournalførTilBehandling(
+                journalpostId = automatiskJournalføringRequest.journalpostId,
+                personIdent = personIdent,
+                stønadstype = stønadstype,
+            )
+        } else {
+            håndterSøknadSomIkkeKanAutomatiskJournalføres()
+        }
+    }
+
+    private fun automatiskJournalførTilBehandling(
+        journalpostId: String,
+        personIdent: String,
+        stønadstype: Stønadstype,
+    ): AutomatiskJournalføringResponse {
+        val journalpost = journalpostService.hentJournalpost(journalpostId)
+        val fagsak = fagsakService.hentEllerOpprettFagsak(personIdent, stønadstype)
+        val nesteBehandlingstype = utledNesteBehandlingstype(behandlingService.hentBehandlinger(fagsak.id))
+        val journalførendeEnhet =
+            arbeidsfordelingService.hentNavEnhetIdEllerBrukMaskinellEnhetHvisNull(fagsak.hentAktivIdent())
+
+        validerKanAutomatiskJournalføre(personIdent, stønadstype, journalpost)
+
+        val behandling = opprettBehandlingOgPopulerGrunnlagsdataForAutomatiskJournalførtSøknad(
+            behandlingstype = nesteBehandlingstype,
+            fagsak = fagsak,
+            søknad = journalpost,
+        )
+
+        journalpostService.oppdaterOgFerdigstillJournalpostMaskinelt(
+            journalpost = journalpost,
+            journalførendeEnhet = journalførendeEnhet,
+            fagsak = fagsak,
+        )
+
+        opprettBehandleSakOppgaveTask(
+            OpprettOppgaveForOpprettetBehandlingTask.OpprettOppgaveTaskData(
+                behandlingId = behandling.id,
+                saksbehandler = SikkerhetContext.hentSaksbehandlerEllerSystembruker(),
+                beskrivelse = "Automatisk journalført søknad",
+            ),
+        )
+        return AutomatiskJournalføringResponse(
+            fagsakId = fagsak.id,
+            behandlingId = behandling.id,
+        )
+    }
+
+    private fun håndterSøknadSomIkkeKanAutomatiskJournalføres(): AutomatiskJournalføringResponse {
+        TODO("Not yet implemented")
+    }
+
+    fun kanOppretteBehandling(ident: String, stønadstype: Stønadstype): Boolean {
+        val allePersonIdenter = personService.hentPersonIdenter(ident).identer()
+        val fagsak = fagsakService.finnFagsak(allePersonIdenter, stønadstype)
+        val behandlinger = fagsak?.let { behandlingService.hentBehandlinger(fagsak.id) } ?: emptyList()
+        val behandlingstype = utledNesteBehandlingstype(behandlinger)
+
+        return when (behandlingstype) {
+            BehandlingType.FØRSTEGANGSBEHANDLING -> true
+            BehandlingType.REVURDERING -> kanAutomatiskJournalføreRevurdering(behandlinger, fagsak)
+        }
```

Comment:
Trenger vi egentlige å finne behandlingstypen? Eller er det tilstrekkelig å sjekke om siste behandling er ferdigstilt eller ikke? 
Kan eventuellt erstatte rad 110-115 med 
```return kanAutomatiskJournalføreRevurdering(behandlinger, fagsak)`

Reply:
Nei, godt poeng. Dette henger nok igjen fra at vi i starten bare automatisk journalførte for førstegangsbehandlinger. Fikser